### PR TITLE
Channel status - get MSP details from Fabric

### PR DESCRIPTION
**Pull request by abhakish**
This api is used for fetching the msp details from fabric to know which all orgs have joined the channel

Signed-off-by: Abhay Kishore <abhay.kishore@walmart.com>

### CODE REVIEW

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java [hunk 1]
```diff
@@ -116,6 +129,35 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
...
+    Configtx.Config channelConfig;
+    try {
+      final byte[] channelConfigurationBytes = channel.getChannelConfigurationBytes();
+      channelConfig = Configtx.Config.parseFrom(channelConfigurationBytes);
+    } catch (Exception e) {
+      log.error("Error in fetching channel members", e);
+      throw new ChannelOperationException(ErrorCode.CHANNEL_MEMBERS_MSPID_FAILED, "get channel members mspid failed",
+          e.getCause());
+    }
+    return channelConfig.getChannelGroup().getGroupsMap().get(GROUP_TAG_APPLICATION).getGroupsMap().keySet();
```
Comment (nithin-pankaj) [inline src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:158]: [nitpick] This is too much chaining. Maybe you can break this into two.

**Review by nithin-pankaj — APPROVED** [review]

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java [hunk 2]
```diff
@@ -116,6 +129,35 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
...
+        log.error("Error in fetching channel members", e);
+        throw new ChannelOperationException(ErrorCode.CHANNEL_MEMBERS_MSPID_FAILED, "get channel members mspid failed",
+            e.getCause());
+      }
+    }
+
+    Configtx.Config channelConfig;
+    try {
+      final byte[] channelConfigurationBytes = channel.getChannelConfigurationBytes();
+      channelConfig = Configtx.Config.parseFrom(channelConfigurationBytes);
```
Comment (nidhi-singh02) [inline src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:152]: Why are we fetching channel config here ?  why are  we getting MSP ID's in two different ways ?

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java [hunk 3]
```diff
@@ -116,6 +129,35 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
...
+    } catch (Exception e) {
+      log.error("Error in fetching channel members", e);
+      throw new ChannelOperationException(ErrorCode.CHANNEL_MEMBERS_MSPID_FAILED, "get channel members mspid failed",
```
Comment (nidhi-singh02) [inline src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:155]: Error message should be more appropriate. As here performed operation is different.

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java (hunk 1 above)
Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java]: Simplified the logic

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java [hunk 4]
```diff
@@ -116,6 +129,14 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
     }
   }
 
+  @Override
+  public Set<String> getChannelMembersMSPID(String channelName) {
```
Comment (nidhi-singh02) [inline src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:133]: Can we check and add test cases for a couple of scenarios -
1) Channel does not exists
2) Channel not being passed or passed as ""(empty string) while calling

Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:133]: 1. If channel doesn't exist then inside this method `gateway.getNetwork(channelName)` a new channel is getting.
2.  Added a check for handling empty string scenario, for channel not being passed that is already handled at controller level.

Comment (nidhi-singh02) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:133]: Can we add test case for case 1. Directly call getChannelMembersMSPIDTest, let's test the response.

Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:133]: testcase added

File: src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java [hunk 5]
```diff
@@ -116,6 +129,17 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
...
+
+    if (StringUtils.isEmpty(channelName)) {
```
Comment (nithin-pankaj) [inline src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java:135]: [nitpick] you can prefer StringUtils#isBlank over isEmpty.
The latter return false for string containing only whitespace `" "`

Comment (abhakish) [reply src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java]: changes done

**Review by arsulegai — APPROVED** [review]

**Pull request merged**